### PR TITLE
SRV-207 Add classifier

### PR DESCRIPTION
**Pull request by PawelPodkalicki**

### CODE REVIEW

File: src/app/common/components/banner-preview/banner-preview.component.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<div class="banner-preview">
+  <ng-container *ngIf="banner.type === 0">
```
Comment (c3zi) [inline src/app/common/components/banner-preview/banner-preview.component.html]: You can use `adTypesEnum` from `ad.enum.ts` file.

File: src/app/publisher/classifier/classifier-list-item/classifier-list-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    this.setClassified(isApproved);
+
+    this.publisherService.setBannerClassification(this.bannerClassification.bannerId, isApproved, this.siteId).subscribe(
+      () => {
+      },
+      (error: HttpErrorResponse) => {
+        this.setClassified(previousClassified);
+        if (error.status !== codes.HTTP_INTERNAL_SERVER_ERROR) {
+          // TODO handle error
+          console.error('ClassifierListItemComponent', error);
```
Comment (c3zi) [inline src/app/publisher/classifier/classifier-list-item/classifier-list-item.component.ts]: We don't need `console.error`, but for sure we need to inform a user about errors :)

File: src/app/publisher/classifier/classifier-list-item/classifier-list-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  ngOnInit(): void {
+    this.isGlobal = this.siteId === null;
+    this.bannerClassification = {
+      ...this.bannerClassification,
+      type: this.adTypesEnum[this.bannerClassification.type.toUpperCase()],
+    }
+  }
+
+  setClassified(isApproved: boolean | null): void {
```
Comment (c3zi) [inline src/app/publisher/classifier/classifier-list-item/classifier-list-item.component.ts]: `setClassificationStatus` ?

File: src/app/publisher/classifier/classifier-list-item/classifier-list-item.component.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    } else {
+      this.bannerClassification.classifiedSite = isApproved;
+    }
+  }
+
+  get classified(): boolean | null {
+    return this.isGlobal ? this.bannerClassification.classifiedGlobal : this.bannerClassification.classifiedSite;
+  }
+
+  setBannerClassification(isApproved: boolean): void {
```
Comment (c3zi) [inline src/app/publisher/classifier/classifier-list-item/classifier-list-item.component.ts]: `classify` ?

File: src/app/publisher/classifier/classifier.component.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    this.publisherService.getBannerClassification(this.siteId, limit, offset)
+      .subscribe(
+        (bannerList: BannerClassification[]) => {
+          this.bannerClassifications = bannerList;
+          this.isLoading = false;
+        },
+        (error: HttpErrorResponse) => {
+          if (error.status !== codes.HTTP_INTERNAL_SERVER_ERROR) {
+            // TODO handle error
+            console.error('ClassifierComponent', error);
```
Comment (c3zi) [inline src/app/publisher/classifier/classifier.component.ts]: As far as I know, we have a dedicated component to display error messages. Could we use it here?

**Comment by PawelPodkalicki** [comment]
Scaling iframes in preview added as task https://adshares.atlassian.net/browse/SRV-231

**Pull request merged**